pkg/agent/ipassigner/responder/arp_responder.go

```diff
@@ -65,40 +56,88 @@ func (r *arpResponder) RemoveIP(ip net.IP) error {
 		return fmt.Errorf("only IPv4 is supported")
 	}
 	if r.deleteIP(ip) {
-		klog.InfoS("Removed IP from ARP responder", "ip", ip, "interface", r.iface.Name)
+		klog.InfoS("Removed IP from ARP responder", "ip", ip, "interface", r.linkName)
 	}
 	return nil
 }
 
-func (r *arpResponder) handleARPRequest() error {
-	pkt, _, err := r.conn.Read()
+func (r *arpResponder) handleARPRequest(client *arp.Client, iface *net.Interface) error {
+	pkt, _, err := client.Read()
 	if err != nil {
 		return err
 	}
 	if pkt.Operation != arp.OperationRequest {
 		return nil
 	}
 	if !r.isIPAssigned(pkt.TargetIP) {
-		klog.V(4).InfoS("Ignored ARP request", "ip", pkt.TargetIP, "interface", r.iface.Name)
+		klog.V(4).InfoS("Ignored ARP request", "ip", pkt.TargetIP, "interface", r.linkName)
 		return nil
 	}
-	if err := r.conn.Reply(pkt, r.iface.HardwareAddr, pkt.TargetIP); err != nil {
+	if err := client.Reply(pkt, iface.HardwareAddr, pkt.TargetIP); err != nil {
 		return fmt.Errorf("failed to reply ARP packet for IP %s: %v", pkt.TargetIP, err)
 	}
-	klog.V(4).InfoS("Sent ARP response", "ip", pkt.TargetIP, "interface", r.iface.Name)
+	klog.V(4).InfoS("Sent ARP response", "ip", pkt.TargetIP, "interface", r.linkName)
 	return nil
 }
 
 func (r *arpResponder) Run(stopCh <-chan struct{}) {
+	// The responder instance is created by the factory and can be shared by multiple callers.
+	// Using once.Do here ensures it is started only once.
+	r.once.Do(func() {
```

Comment:
`r.Once.Do` is not really meant to be used with a blocking function
subsequent calls to `Run` will remain blocked inside `r.Once.Do` for the duration of the program, which I think is a bit misleading
It may be worth starting a goroutine instead
```go
	r.once.Do(func() {
		go wait.NonSlidingUntil(func() {
			r.dialAndHandleRequests(stopCh)
		}, time.Second, stopCh)
	})
	<-stopCh

Reply:
done. Thanks!